ringo_printtemplate/odfconv.py

```diff
@@ -67,6 +68,10 @@ def convert(self, data, format="ods", update=True):
         :returns: Converted data.
 
         """
+        # check document type before writing or opening
+        mimetype = check_mime_type_from_buffer(data)
+        if mimetype != "application/vnd.oasis.opendocument.text":
+            return None
```

Comment:
This check will never fail and has therefor no effect. This is because calling the converter is always called at the end of the renderering process in the render_final_response method. So it will operate on a Document which has already been checked in _render_response/render_final_response.